src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -176,6 +176,15 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleSwitchView(commandResult.getTargetView().trim());
             }
 
+            if (commandResult.isUndo()) {
+                logic.getModel().setState((ElisaStateManager)
+                        logic.getModel().getElisaStateHistory().peekCommand().deepCopy());
+            } else {
+                logic.getModel().getElisaStateHistory().pushCommand(logic.getModel().getState().deepCopy());
+            }
+
+            logic.getModel().updateModelLists();
+
```

Comment:
Is there any reason why you would choose to do this within the Ui instead of at LogicManager#execute? Right now I think it should be done at the LogicManager as this functionality does not belong to the Ui.